File: PRNumberBiz/app/src/main/java/com/pplus/prnumberbiz/apps/customer/ui/RecommendActivity.java
package com.pplus.prnumberbiz.apps.customer.ui;

import android.content.Intent;
import android.net.Uri;
import android.provider.Telephony;
import androidx.annotation.NonNull;
import android.os.Bundle;
import android.view.View;
import android.widget.TextView;

import com.pple.pplus.utils.part.info.OsUtil;
import com.pple.pplus.utils.part.utils.StringUtils;
import com.pplus.prnumberbiz.Const;
import com.pplus.prnumberbiz.R;
import com.pplus.prnumberbiz.apps.common.mgmt.CountryConfigManager;
import com.pplus.prnumberbiz.apps.common.mgmt.LoginInfoManager;
import com.pplus.prnumberbiz.apps.common.toolbar.ImplToolbar;
import com.pplus.prnumberbiz.apps.common.toolbar.OnToolbarListener;
import com.pplus.prnumberbiz.apps.common.toolbar.ToolbarOption;
import com.pplus.prnumberbiz.apps.common.ui.base.BaseActivity;

public class RecommendActivity extends BaseActivity implements ImplToolbar{

    @Override
    public String getPID(){

        return null;
    }

    @Override
    public int getLayoutRes(){

        return R.layout.activity_recommend;
    }

    private String mMobileNumber;

    @Override
    public void initializeView(Bundle savedInstanceState){

        mMobileNumber = getIntent().getStringExtra(Const.MOBILE_NUMBER);
        TextView text_recommend = (TextView) findViewById(R.id.text_recommend_code);
        text_recommend.setText("" + LoginInfoManager.getInstance().getUser().getRecommendKey());

        TextView text_cash = (TextView)findViewById(R.id.text_recommend_cash);
        text_cash.setText(getString(R.string.format_msg_recommend_description2, CountryConfigManager.getInstance().getConfig().getProperties().getRecommendBol()));

        findViewById(R.id.layout_recommend_sms).setOnClickListener(new View.OnClickListener(){

            @Override
            public void onClick(View view){

                String smsBody = getString(R.string.msg_invite_recommend);

                if(OsUtil.isKitkat()) //At least KitKat
                {
                    String defaultSmsPackageName = Telephony.Sms.getDefaultSmsPackage(RecommendActivity.this); //Need to change the build to API 19

                    Intent sendIntent = new Intent(Intent.ACTION_SEND);
                    sendIntent.setType("text/plain");
                    if(StringUtils.isNotEmpty(mMobileNumber)) {
                        sendIntent.putExtra("address", mMobileNumber);
                    }

                    sendIntent.putExtra(Intent.EXTRA_TEXT, smsBody);

                    if(defaultSmsPackageName != null)//Can be null in case that there is no default, then the user would be able to choose any app that support this intent.
                    {
                        sendIntent.setPackage(defaultSmsPackageName);
                    }
                    startActivity(sendIntent);

                } else //For early versions, do what worked for you before.
                {
                    Intent sendIntent = new Intent(Intent.ACTION_VIEW);
                    sendIntent.setData(Uri.parse("sms:"));
                    if(StringUtils.isNotEmpty(mMobileNumber)) {
                        sendIntent.putExtra("address", mMobileNumber);
                    }
                    sendIntent.putExtra("sms_body", smsBody);
                    startActivity(sendIntent);
                }
            }
        });

        findViewById(R.id.layout_recommend_kakao).setOnClickListener(new View.OnClickListener(){

            @Override
            public void onClick(View view){

//                KakaoLinkUtil.getInstance(RecommendActivity.this).sendKakaoApp(getString(R.string.msg_invite_recommend), "", "");
            }
        });
    }

    @NonNull
    @Override
    public ToolbarOption getToolbarOption(){

        ToolbarOption toolbarOption = new ToolbarOption(this);
        toolbarOption.initializeDefaultToolbar(getString(R.string.msg_recommend), ToolbarOption.ToolbarMenu.RIGHT);
        return toolbarOption;
    }

    @Override
    public OnToolbarListener getOnToolbarClickListener(){

        return new OnToolbarListener(){

            @Override
            public void onClick(View v, ToolbarOption.ToolbarMenu toolbarMenu, Object tag){

                switch (toolbarMenu) {
                    case RIGHT:
                        if(tag.equals(1)) {
                            onBackPressed();
                        }
                        break;
                }
            }
        };

    }
}

File: LuckyBol/app/src/main/java/com/pplus/luckybol/apps/recommend/ui/RecommendActivity.kt
package com.pplus.luckybol.apps.recommend.ui

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import com.pplus.luckybol.R
import com.pplus.luckybol.apps.common.mgmt.CountryConfigManager
import com.pplus.luckybol.apps.common.mgmt.LoginInfoManager
import com.pplus.luckybol.apps.common.toolbar.ImplToolbar
import com.pplus.luckybol.apps.common.toolbar.OnToolbarListener
import com.pplus.luckybol.apps.common.toolbar.ToolbarOption
import com.pplus.luckybol.apps.common.ui.base.BaseActivity
import com.pplus.luckybol.core.util.ToastUtil
import com.pplus.luckybol.databinding.ActivityRecommendBinding
import com.pplus.utils.part.format.FormatUtil

class RecommendActivity : BaseActivity(), ImplToolbar {
    override fun getPID(): String {
        return "Main_number_invite"
    }

    private lateinit var binding: ActivityRecommendBinding

    override fun getLayoutView(): View {
        binding = ActivityRecommendBinding.inflate(layoutInflater)
        return binding.root
    }

    override fun initializeView(savedInstanceState: Bundle?) {

        val configProperties = CountryConfigManager.getInstance().config.properties

        val recommendKey = LoginInfoManager.getInstance().user.recommendKey

        binding.textRecommendKey.text = recommendKey

        binding.textRecommendInvite.setOnClickListener {
            share()
        }

        binding.imageRecommendCopy.setOnClickListener {
            val clipboard = getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager

            val clip = ClipData.newPlainText(getString(R.string.word_recommend_code), recommendKey)
            clipboard.setPrimaryClip(clip)
            ToastUtil.show(this, R.string.msg_copied_clipboard)
        }
    }

    private fun share() {
        val recommendKey = LoginInfoManager.getInstance().user.recommendKey
        val text = "${getString(R.string.format_invite_description, recommendKey)}\n${getString(R.string.format_msg_invite_url, LoginInfoManager.getInstance().user.recommendKey)}"
//        val text = "${getString(R.string.msg_invite_desc)}\n${getString(R.string.msg_invite_url)}"
        val intent = Intent(Intent.ACTION_SEND)
        intent.type = "text/plain"
        intent.putExtra(Intent.EXTRA_TEXT, text)
        val chooserIntent = Intent.createChooser(intent, getString(R.string.word_share))
        startActivity(chooserIntent)
    }

    override fun getToolbarOption(): ToolbarOption {

        val toolbarOption = ToolbarOption(this)
        toolbarOption.initializeDefaultToolbar(getString(R.string.word_invite), ToolbarOption.ToolbarMenu.LEFT)
        toolbarOption.setToolbarMenu(ToolbarOption.ToolbarMenu.RIGHT, getString(R.string.word_recommend_history))
        return toolbarOption
    }

    override fun getOnToolbarClickListener(): OnToolbarListener {

        return OnToolbarListener { v, toolbarMenu, tag ->
            when (toolbarMenu) {
                ToolbarOption.ToolbarMenu.LEFT -> if (tag == 1) {
                    onBackPressed()
                }
                ToolbarOption.ToolbarMenu.RIGHT -> if (tag == 1) {
                    val intent = Intent(this, RecommendHistoryActivity::class.java)
                    intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP
                    startActivity(intent)
                }
            }
        }
    }
}
